Adding pv.kubernetes.io annotations in labels-annotations-taints (#37928)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

{{< note >}} Starting in v1.17, this label is deprecated in favor of [topology.kubernetes.io/zone](#topologykubernetesiozone). {{< /note >}}

### pv.kubernetes.io/bind-completed {#pv-kubernetesiobind-completed}

Example : `pv.kubernetes.io/bind-completed: "yes"`

Used on: PersistentVolumeClaim

> It indicates that the lifecycle of the PVC has passed through the initial setup. This information changes how we interpret some observations of the state of the objects. Value of this Annotation does not matter.

Reviewer: How about:
```suggestion
When this annotation is set on a PersistentVolumeClaim (PVC), that indicates that the
lifecycle of the PVC has passed through initial binding setup. When present, that
information changes how the control plane interprets the state of PVC objects.
The value of this annotation does not matter to Kubernetes.
```

Author: Thanks for the suggestions. I'll make the changes.

> When this annotation is set on a PersistentVolumeClaim (PVC), that indicates that the lifecycle of the PVC has passed through initial binding setup. When present, that information changes how the control plane interprets the state of PVC objects. The value of this annotation does not matter to Kubernetes.

Reviewer 2: ```suggestion
When this annotation is set on a PersistentVolumeClaim (PVC), that indicates that the lifecycle
of the PVC has passed through initial binding setup. When present, that information changes
how the control plane interprets the state of PVC objects.
The value of this annotation does not matter to Kubernetes.
```

### pv.kubernetes.io/bound-by-controller {#pv-kubernetesioboundby-controller}

Example : `pv.kubernetes.io/bound-by-controller: "yes"`

Used on: PersistentVolume, PersistentVolumeClaim

> It indicates that the binding (PV->PVC or PVC->PV) was installed by the controller. The absence of this annotation means the binding was done by the user. Value of this annotation does not matter.

Reviewer: How about this?
```suggestion
If this annotation is set on a PersistentVolume or PersistentVolumeClaim, it
indicates that a storage binding (PersistentVolume 🢡 PersistentVolumeClaim,
or PersistentVolumeClaim 🢡 PersistentVolume) was installed by a
{{< glossary_tooltip text="controller" term_id="controller" >}}.
If the annotation isn't set, and there is a storage binding in place, the absence
of that annotation means that the binding was done manually.
The value of this annotation does not matter; 
```

### pv.kubernetes.io/provisioned-by {#pv-kubernetesiodynamically-provisioned}

> Example : `"pv.kubernetes.io/provisioned-by":"kubernetes.io/rbd"`

Reviewer: ```suggestion
Example : `pv.kubernetes.io/provisioned-by: "kubernetes.io/rbd"`
```

> If this annotation is set on a PersistentVolume or PersistentVolumeClaim , it indicates that a storage binding(PersistentVolume -> PersistentVolumeClaim, or PersistentVolumeClaim -> PersistentVolume) was installed by the {{< glossary_tooltip text="controller" term_id="controller" >}}.If the annotation isn't set, and there is a storage binding in place, the absence of that annotation means that the binding was done manually. The value of this annotation does not matter.

Reviewer 2: ```suggestion
If this annotation is set on a PersistentVolume or PersistentVolumeClaim, it indicates that a storage binding
(PersistentVolume -> PersistentVolumeClaim, or PersistentVolumeClaim -> PersistentVolume) was installed
by the {{< glossary_tooltip text="controller" term_id="controller" >}}.
If the annotation isn't set, and there is a storage binding in place, the absence of that annotation means that
the binding was done manually. The value of this annotation does not matter.
```

Used on: PersistentVolume

> It is added to a PV that has been dynamically provisioned by Kubernetes. Its value is name of volume plugin that created the volume. It serves both user (to show where a PV comes from) and Kubernetes (to recognize dynamically provisioned PVs in its decisions).

Reviewer: Is this an annotation or a label? (we should make it clear to readers).

> This annotation is added to a PersistentVolume(PV) that has been dynamically provisioned by Kubernetes. Its value is the name of volume plugin that created the volume. It serves both user (to show where a PV comes from) and Kubernetes (to recognize dynamically provisioned PVs in its decisions).

Reviewer 2: ```suggestion
This annotation is added to a PersistentVolume(PV) that has been dynamically provisioned by Kubernetes.
Its value is the name of volume plugin that created the volume. It serves both user (to show where a PV
comes from) and Kubernetes (to recognize dynamically provisioned PVs in its decisions).
```

### pv.kubernetes.io/migrated-to {#pv-kubernetesio-migratedto}

Example: `pv.kubernetes.io/migrated-to: pd.csi.storage.gke.io`

Used on: PersistentVolume, PersistentVolumeClaim

> It is added to a PersistentVolume(PV) and PersistentVolumeClaim(PVC) that is supposed to be dynamically provisioned/deleted by its corresponding CSI driver through the CSIMigration feature flags. When this annotation is set, the Kubernetes components will "stand-down" and the external-provisioner will act on the objects.

Reviewer 2: ```suggestion
It is added to a PersistentVolume(PV) and PersistentVolumeClaim(PVC) that is supposed to be
dynamically provisioned/deleted by its corresponding CSI driver through the `CSIMigration` feature gate.
When this annotation is set, the Kubernetes components will "stand-down" and the `external-provisioner`
will act on the objects.
```

{{< note >}} Starting in v1.17, this label is deprecated in favor of [topology.kubernetes.io/zone](#topologykubernetesiozone). {{< /note >}}

### pv.kubernetes.io/bind-completed {#pv-kubernetesiobind-completed}

Example: `pv.kubernetes.io/bind-completed: "yes"`

Used on: PersistentVolumeClaim

When this annotation is set on a PersistentVolumeClaim (PVC), that indicates that the lifecycle
of the PVC has passed through initial binding setup. When present, that information changes
how the control plane interprets the state of PVC objects.
The value of this annotation does not matter to Kubernetes.

### pv.kubernetes.io/bound-by-controller {#pv-kubernetesioboundby-controller}

Example: `pv.kubernetes.io/bound-by-controller: "yes"`

Used on: PersistentVolume, PersistentVolumeClaim

If this annotation is set on a PersistentVolume or PersistentVolumeClaim, it indicates that a storage binding

> (PersistentVolume -> PersistentVolumeClaim, or PersistentVolumeClaim -> PersistentVolume) was installed

Reviewer: ```suggestion
(PersistentVolume → PersistentVolumeClaim, or PersistentVolumeClaim → PersistentVolume) was installed
```

This can help with accessibility: a screen reader should always know that → sounds like “right arrow”. I haven't tested this thinking thoroughly though.

by the {{< glossary_tooltip text="controller" term_id="controller" >}}.
If the annotation isn't set, and there is a storage binding in place, the absence of that annotation means that
the binding was done manually. The value of this annotation does not matter.

### pv.kubernetes.io/provisioned-by {#pv-kubernetesiodynamically-provisioned}

> Example: `pv.kubernetes.io/provisioned-by:"kubernetes.io/rbd"`

Reviewer: (nit)
```suggestion
Example: `pv.kubernetes.io/provisioned-by: "kubernetes.io/rbd"`
```

It's OK to merge this PR even if this isn't fixed.
